vector.mli

```diff
@@ -43,41 +42,48 @@ type 'a t
 (** {2 Operations proper to vectors, or with a different type and/or
     semantics than those of module [Array]} *)
 
-val make: int -> dummy:'a -> 'a t
-(** [Vector.make n dummy] returns a fresh vector of length [n].
+val make: int -> 'a -> 'a t
+(** [Vector.make n v] returns a fresh vector of length [n].
    All the elements of this new vector are initially
-   physically equal to [dummy] (in the sense of the [==] predicate).
+   physically equal to [v] (in the sense of the [==] predicate).
 
-   Raise [Invalid_argument] if [n < 0] or [n > Sys.max_array_length].
-   If the value of [dummy] is a floating-point number, then the maximum
-   size is only [Sys.max_array_length / 2].*)
+   Raise [Invalid_argument] if [n < 0] or [n >= Sys.max_array_length]. *)
 
-val create: dummy:'a -> 'a t
-(** [Vector.create dummy] returns a fresh vector of length [0]. *)
+val create: unit -> 'a t
+(** [Vector.create ()] returns a fresh vector of length [0]. *)
 
-val init: int -> dummy:'a -> (int -> 'a) -> 'a t
+val init: int -> (int -> 'a) -> 'a t
 (** [Vector.init n f] returns a fresh vector of length [n],
    with element number [i] initialized to the result of [f i].
    In other terms, [Vector.init n f] tabulates the results of [f]
    applied to the integers [0] to [n-1].
 
-   Raise [Invalid_argument] if [n < 0] or [n > Sys.max_array_length].
-   If the return type of [f] is [float], then the maximum
-   size is only [Sys.max_array_length / 2].*)
+   Raise [Invalid_argument] if [n < 0] or [n >= Sys.max_array_length]. *)
 
-val resize: 'a t -> int -> unit
-(** [Vector.resize a n] sets the length of vector [a] to [n].
+val resize: 'a t -> int -> 'a -> unit
+(** [Vector.resize a n v] sets the length of vector [a] to [n].
+    If [n > Vector.length a], the new elements in the vector are
+    initially physically equal to [v].
 
    The elements that are no longer part of the vector, if any, are
-   internally replaced by the dummy value of vector [a], so that they
+   cleared from the internal representation, so that they
    can be garbage collected when possible.
 
-   Raise [Invalid_argument] if [n < 0] or [n > Sys.max_array_length]. *)
+   Raise [Invalid_argument] if [n < 0] or [n >= Sys.max_array_length]. *)
 
+val shrink: 'a t -> int -> unit
+(** [Vector.shrink a n] sets the length of vector [a] to be at most [n].
+
+    If [n >= Vector.length a], this operation has no effect.
```

Comment:
I'm unsure if this is the ideal semantics for `shrink` (or if the name is confusing given those semantics). Perhaps the user will expect "`shrink a n` ensures `length a = n`", in which case perhaps we should raise here too.

Reply:
Yes, I have mixed feelings about this one.
* If `shrink` is exposed in the API, I would prefer it to fail if `n > Vector.length a`
* A simpler solution is not to expose it, since users can always use `resize` to implement their own `shrink` on top of it.